src/util/types.ts

```diff
@@ -53,7 +69,7 @@ export interface ClientFile {
     readonly archiveFilename: string,
     readonly archiveFileUrl: string,
     readonly clientName: InstallableClient,
-    readonly directoryUrl: string,
+    readonly directoryUrl?: string,
```

Comment:
I have taken `directoryUrl` as the paramter that can tell if the particular cllient is downloaded from `github` or from `mirror`. As there is no such directory url in case of `github`